sklearn/feature_selection/_univariate_selection.py

```diff
@@ -351,12 +385,28 @@ def f_regression(X, y, *, center=True):
     SelectPercentile: Select features based on percentile of the highest
         scores.
     """
-    correlation_coefficient = r_regression(X, y, center=center)
+    correlation_coefficient = r_regression(
+        X, y, center=center, force_finite=force_finite
+    )
     deg_of_freedom = y.size - (2 if center else 1)
 
     corr_coef_squared = correlation_coefficient ** 2
-    f_statistic = corr_coef_squared / (1 - corr_coef_squared) * deg_of_freedom
-    p_values = stats.f.sf(f_statistic, 1, deg_of_freedom)
+
+    with np.errstate(divide="ignore", invalid="ignore"):
+        f_statistic = corr_coef_squared / (1 - corr_coef_squared) * deg_of_freedom
+        p_values = stats.f.sf(f_statistic, 1, deg_of_freedom)
+
+    if force_finite and not np.isfinite(f_statistic.sum()):
```

Comment:
Nit: No strong opinion, but I think this is slightly clearer:

```suggestion
    if force_finite and not np.isfinite(f_statistic).any():
```

Reply:
you should mean `all`